src/views/orderReport.tsx

```diff
@@ -0,0 +1,467 @@
+        backgroundColor: [
+          "rgb(0, 143, 251)",
+          "rgb(0, 227, 150)",
+          "rgb(254, 176, 25)",
+          "rgb(255, 69, 96)",
+          "rgb(119, 93, 208)",
+          "rgb(0, 143, 251)",
+          "rgb(0, 227, 150)",
+          "rgb(254, 176, 25)",
+          "rgb(255, 69, 96)",
+          "rgb(119, 93, 208)",
+          "rgb(0, 143, 251)",
+          "rgb(0, 143, 251)",
+          "rgb(0, 227, 150)",
+          "rgb(254, 176, 25)",
+          "rgb(255, 69, 96)",
+          "rgb(119, 93, 208)"
+        ],
+        hoverBackgroundColor: [
+          "rgb(0, 143, 251, 0.6)",
+          "rgb(0, 227, 150, 0.6)",
+          "rgb(254, 176, 25, 0.6)",
+          "rgb(255, 69, 96, 0.6)",
+          "rgb(119, 93, 208, 0.6)",
+          "rgb(0, 143, 251, 0.6)",
+          "rgb(0, 227, 150, 0.6)",
+          "rgb(254, 176, 25, 0.6)",
+          "rgb(255, 69, 96, 0.6)",
+          "rgb(119, 93, 208, 0.6)",
+          "rgb(0, 143, 251, 0.6)",
+          "rgb(0, 143, 251, 0.6)",
+          "rgb(0, 227, 150, 0.6)",
+          "rgb(254, 176, 25, 0.6)",
+          "rgb(255, 69, 96, 0.6)",
+          "rgb(119, 93, 208, 0.6)"
+        ],
+        borderColor: "#161A1D",
+        borderWidth: 1,
+      },
+    ]
+    let chartData = { labels: ORDER_REPORT_CHART_LABELS.PAYMENTWISE_ORDER_LABELS, datasets: datasets };
+    setPaymentWiseOrderData(chartData);
+  }
+
+  const getHourlyTrendData = () => {
+    let data = hourlyReport.hourlyHistoricalOrderData;
+    let todayData = {
+      label: HOURLY_TREND_DATA_LABELS.TODAY,
+      data: data.today,
+      borderColor: 'rgb(0, 143, 251)'
+    };
+
+    let yesterdayData = {
+      label: HOURLY_TREND_DATA_LABELS.YESTERDAY,
+      data: data.yesterday,
+      borderColor: 'rgb(0, 227, 150)'
+    };
+    let lastWeekData = {
+      label: HOURLY_TREND_DATA_LABELS.LAST_WEEK,
+      data: data.lastWeek,
+      borderColor: 'rgb(254, 176, 25)'
+    };
+    let lastMonthData = {
+      label: HOURLY_TREND_DATA_LABELS.LAST_MONTH,
+      data: data.lastMonth,
+      borderColor: 'rgb(255, 69, 96)'
+    };
+    let lastYearData = {
+      label: HOURLY_TREND_DATA_LABELS.LAST_YEAR,
+      data: data.lastYear,
+      borderColor: 'rgb(119, 93, 208)'
+    };
+
+    const chartData = {
+      labels: ORDER_REPORT_CHART_LABELS.HOURLY_TIMESTAMPS,
+      datasets: [todayData, yesterdayData, lastWeekData, lastMonthData, lastYearData]
+    };
+    setHourlyOrderTrendData(chartData);
+  }
+
+  const getLocaleWiseHourlyOrderData = () => {
+    let slicedOrderData = hourlyReport.hourlySlicedOrderData;
+    let historicalOrderData = hourlyReport.hourlyHistoricalOrderData;
+    let summary = hourlyReport.summary;
+    let time = ORDER_REPORT_CHART_LABELS.HOURLY_TIMESTAMPS.slice(1);
+    let totalData = [getTotalData(historicalOrderData.lastYear),
```

Comment:
always declare a method first before invoking it.
Here getTotalData is being declared after this method.